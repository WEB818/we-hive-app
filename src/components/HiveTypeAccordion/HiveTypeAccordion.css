.Accordion__section {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: flex-start;
  width: 100vw;
}

.Accordion {
  background: -webkit-linear-gradient(
    top,
    rgba(252, 234, 187, 1) 0%,
    rgba(248, 181, 0, 0.83) 83%,
    rgba(251, 223, 147, 0.8) 100%
  );
  color: #1a1c24;
  cursor: pointer;
  padding: 18px;
  display: flex;
  align-items: center;
  border: none;
  outline: none;
  transition: background-color 0.6s ease;
}

.Accordion:hover,
.active {
  background-color: #ccc;
}

.Accordion__title {
  font-family: "Open Sans", sans-serif;
  font-weight: 600;
  font-size: 14px;
}

/* .Accordion__icon {
  margin-left: auto;
  transition: transform 0.6s ease;
} */

/* .rotate {
  transform: rotate(90deg);
} */

.Accordion__goal {
  background-color: white;
  overflow: hidden;
  transition: max-height 0.6s ease;
}
