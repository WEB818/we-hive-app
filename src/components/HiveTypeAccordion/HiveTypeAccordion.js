import React, { Component } from "react";
import HiveContext from "../../context/HiveContext";
import { Button } from "../../components/Utils/Utils";
import AccordionItem from "../../components/AccordionItem/AccordionItem";
import "./HiveTypeAccordion.css";

export default class HiveTypeAccordion extends Component {
  static defaultProps = {
    goals: []
  };

  state = {
    expanded: false
  };

  static contextType = HiveContext;

  handleSetActiveType = () => {
    this.setState(prevState => ({
      expanded: !prevState.expanded
    }));
  };

  renderButtons(expanded) {
    const { type } = this.props;
    return (
      <div className="Accordion__section">
        <Button
          className="Accordion"
          onClick={() => this.handleSetActiveType()}
        >
          <p className="Accordion__title">{type}</p>
        </Button>
        {expanded && this.renderTitles()}
      </div>
    );
  }

  renderTitles() {
    const { goals, id } = this.props;
    const goalFilter = goals.filter(goal => goal.goal_type === id);

    return (
      <>
        <ul className="HiveType__list">
          <li className="goal-desc">
            {!goalFilter.length && (
              <AccordionItem goalTitle="No hives yet" id="" />
            )}
          </li>
        </ul>
        <ul className="HiveType__list">
          {goalFilter.map(goal => (
            <li key={goal.id} className="goal-desc">
              <AccordionItem id={goal.id} goalTitle={goal.goal_description} />
            </li>
          ))}
        </ul>
      </>
    );
  }

  render() {
    const { expanded } = this.state;

    return <>{this.renderButtons(expanded)}</>;
  }
}
