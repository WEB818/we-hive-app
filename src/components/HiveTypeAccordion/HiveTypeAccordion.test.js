import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import HiveTypeAccordion from "./HiveTypeAccordion";

describe(`HiveTypeAccordion Component`, () => {
  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<HiveTypeAccordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("opens any clicked tab", () => {
    const wrapper = shallow(<HiveTypeAccordion />);
    wrapper
      .find(".Accordion")
      .at(0)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
