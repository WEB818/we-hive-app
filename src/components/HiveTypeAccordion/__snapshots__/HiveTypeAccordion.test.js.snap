// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiveTypeAccordion Component opens any clicked tab 1`] = `
<Fragment>
  <div
    className="Accordion__section"
  >
    <Button
      className="Accordion"
      onClick={[Function]}
    >
      <p
        className="Accordion__title"
      />
    </Button>
    <ul
      className="HiveType__list"
    >
      <li
        className="goal-desc"
      >
        <AccordionItem
          goalTitle="No hives yet"
          id=""
        />
      </li>
    </ul>
    <ul
      className="HiveType__list"
    />
  </div>
</Fragment>
`;
